--- VRC-YTdl-AutoUpdater_old.ps1	2019-09-01 23:43:00.000000000 +0900
+++ VRC-YTdl-AutoUpdater_new.ps1	2020-01-26 19:09:28.000000000 +0900
@@ -29,14 +29,41 @@
     exit
 }
 
+# MD5チェックサムを取得
+. $PSScriptRoot"\VRC-YTdl-CheckSumValidator.ps1"
+$hashFromWeb = Get-YTdlRemoteHash -ParsedJsonOfYoutubeDlReselaseUrl $responce -CheckSumAssetName "MD5SUMS"
+if($hashFromWeb -eq $null){
+    Write-Host "リモートにチェックサムが見つかりませんでした。"
+    #$null = Read-Host '終了するには、Enter キーを押してください...'
+    exit
+}
+Write-Host "リモートのチェックサム: "$hashFromWeb
+$newExeName = "new_" + "$exeName"
+
 #違ったら更新
 Move-Item $exeName "bak_youtube-dl.exe" -Force
 $downloadUrl = $responce[0].assets.browser_download_url | sls "youtube-dl.exe$"
 $uri = New-Object Uri($downloadUrl)
 $dlPath = Get-Location
-$dlPath = "$dlPath"+"\"+"$exeName"
+# $dlPath = "$dlPath"+"\"+"$exeName"
+$dlPath = "$dlPath"+"\"+"$newExeName" #チェックサムを比較するため別名でダウンロード 
 $cli = New-Object System.Net.WebClient
 $cli.DownloadFile($uri, $dlPath)
+
+#チェックサムを算出
+$fileHash = Get-FileHash -Path $dlPath -Algorithm MD5
+$hashFromFile = $fileHash.Hash
+#チェックサムを比較
+if($hashFromFile -ne $hashFromWeb){
+    Remove-Item $newExeName
+    Write-Host "ファイルのチェックサム"$hashFromFile
+    Write-Host "リモートのチェックサムと一致しませんでした。"
+    #$null = Read-Host '終了するには、Enter キーを押してください...'
+    exit
+}
+#チェックサムが一致したら更新
+Move-Item $newExeName $exeName -Force
+
 Write-Host $downloadUrl
 Write-Host "更新しました"
 #$null = Read-Host '終了するには、Enter キーを押してください...'
